src/textual/widget.py

```diff
@@ -2127,11 +2146,16 @@ def post_render(self, renderable: RenderableType) -> ConsoleRenderable:
     def watch_mouse_over(self, value: bool) -> None:
         """Update from CSS if mouse over state changes."""
         if self._has_hover_style:
-            self.app.update_styles(self)
+            self._update_styles()
 
     def watch_has_focus(self, value: bool) -> None:
         """Update from CSS if has focus state changes."""
-        self.app.update_styles(self)
+        self._update_styles()
+
+    def watch_disabled(self) -> None:
+        """Update the styles of the widget and its children when disabled is toggled."""
+        for node in self.walk_children(with_self=True):
```

Comment:
I don't think this is necessary as `app.update_styles` will also update styles for its children.